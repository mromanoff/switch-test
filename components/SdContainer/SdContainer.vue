<template functional>
  <div class="SdContainer" v-bind="data.attrs">
    <div class="SdContainer-main">
      <slot name="main" />
    </div>
  </div>
</template>

<script>
export default {
  props: {
    isTop: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style scoped>
.SdContainer {
  display: flex;
  flex-direction: column;
  margin-left: auto;
  margin-right: auto;
  padding: 0 var(--space--medium);
  width: 100%;
}

.SdContainer-main {
  flex: 1 1 100%;
  width: 100%;
}

@media (--viewport--medium) {
  .SdContainer-main {
    margin-bottom: 0;
    display: block;
  }
}

@media (--viewport--large-up) {
  .SdContainer {
    flex-direction: row;
    max-width: var(--breakpoint--xLarge);
  }
}
</style>
