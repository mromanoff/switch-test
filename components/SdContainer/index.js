import SdContainer from './SdContainer.vue';

export default SdContainer;
