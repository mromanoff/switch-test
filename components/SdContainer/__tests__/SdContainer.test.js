import { shallowMount } from '@vue/test-utils';
import SdContainer from '../SdContainer.vue';

describe('SdContainer.vue', () => {
  let wrapper;

  it('renders default SdContainer', () => {
    wrapper = shallowMount(SdContainer);

    expect(wrapper).toMatchSnapshot();
  });
});
