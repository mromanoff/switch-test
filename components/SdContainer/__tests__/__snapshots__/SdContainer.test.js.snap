// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SdContainer.vue renders default SdContainer 1`] = `
Wrapper {
  "isFunctionalComponent": true,
}
`;
